### PR TITLE
Fixes based on DX test feeback

### PR DESCRIPTION
**Pull request by pleku**
New API, improved javadocs.

### CODE REVIEW

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java [hunk 1]
```diff
@@ -2407,11 +2413,25 @@ public void setItems(Stream<T> streamOfItems) {
...
 
+    /**
+     * Gets the lazy data view for the grid. This data view should only be used
+     * when the used data source is of lazy type and set with:
+     * <ul>
+     * <li>{@link #setDataSource(CallbackDataProvider.FetchCallback)}</li>
+     * <li>{@link #setDataSource(CallbackDataProvider.FetchCallback, CallbackDataProvider.CountCallback)}</li>
+     * <li>{@link #setDataSource(BackEndDataProvider)}</li>
+     * </ul>
+     * If the data source is of wrong type (in-memory), and exception is thrown.
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java:2430]: Done

File: vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/dataview/AbstractUndefinedSizeGridPage.java [hunk 2]
```diff
@@ -122,13 +116,13 @@ public AbstractUndefinedSizeGridPage() {
     private void initNavigationLinks() {
         menuBar.add("Open initially with");
         menuBar.add(
-                new RouterLink("UndefinedSize", UndefinedSizeGridPage.class));
+                new RouterLink("UndefinedSize", RowCountUnknownGridPage.class));
```
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/dataview/AbstractUndefinedSizeGridPage.java:119]: Done

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java [hunk 3]
```diff
@@ -2353,6 +2358,7 @@ public void setItems(Collection<T> items) {
 
     /**
      * {@inheritDoc}
+     * 
      * @deprecated use {@link HasListDataView#setDataSource(Stream)}
```
Comment (mshabarov) [inline vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java:2362]:
```suggestion
     * @deprecated use {@link HasListDataView#setDataSource(Collection)}
```

**Comment by mshabarov** [comment]
No `setDataSource(Stream)` method, thus the build is failing.

**Pull request merged**